### PR TITLE
expanding metric overview in observability pillars

### LINKED ISSUE #99: Edits for Metrics section of Pillars
While https://github.com/microsoft/code-with-engineering-playbook/tree/master/observability#metrics had a good first run of content, it could be formatted/edited in a more structured way.

Proposal for the structure of the 3 sections of the Observability Pillars:
- Summarize what the particular pillar is and how it fits into a dev/ops workflow
- Point out primary use case examples
- Best Practices & Guidance for the particular pillar (may expand into its own doc over time)
- Links To Useful Tools & Further Guidance

### PR DESCRIPTION
**Pull request by khaines**
Expansion and edits for the metrics overview in the observability pillars.

Closes #99 .

### CODE REVIEW

File: observability/pillars.md [hunk 1]
```diff
@@ -47,8 +47,32 @@ Produces the information required to observe series of correlated operations in
 
 ## Metrics
 
-Metrics provide a near real-time indication of how the system is running. As opposed to logs and traces, the amount of data collected using metrics remains constant as the system load increases.
+Metrics provide a near real-time stream of data, informing operators and stakeholders about the functions the system is performing as well as its health. Unlike logging & tracing, metric data tends to be more efficient to transmit and store.
```
Comment (TessFerrandez) [inline observability/pillars.md]: about the functions the system is performing -> about how the functions in the system are performing

**Comment by NarmathaBala** [comment]
The "what should be tracked" section with specific examples is a great idea. 
If you have specific "Do's" and "Don't" with Metrics, tracing  that you may want to add to "Recommendations and What to watch for" sections - it will be a great add (as a part of a separate PR)
NIT: replace & with 'and'

**Comment by fnocera** [comment]
Please reference the issue this PR is addressing

File: observability/pillars.md (hunk 1 above)
Comment (khaines) [reply observability/pillars.md]: Thanks for the feedback here. This part was purposefully about what the component/system is doing (performing an action), vs how well the action was performed. Both are part observability measures, and I'll find a way in a future update to incorporate perf measurement wording in the summary.

**Comment by khaines** [comment]
> The "what should be tracked" section with specific examples is a great idea.
> If you have specific "Do's" and "Don't" with Metrics, tracing that you may want to add to "Recommendations and What to watch for" sections - it will be a great add (as a part of a separate PR)
> NIT: replace & with 'and'

Thank you @NarmathaBala. Yes, I will be adding some do's and don'ts in the next segment of edits I'm going to take on.

**Pull request merged**